src/query.rs

```diff
@@ -147,7 +147,7 @@ impl Query {
         let use_post = !read_only || query.len() > MAX_QUERY_LEN_TO_USE_GET;
 
         let (method, body, content_length) = if use_post {
-            if read_only {
+            if read_only && !self.client.options.contains_key("readonly") {
```

Comment:
What if `read-only=0` is provided? According to https://clickhouse.com/docs/en/operations/settings/permissions-for-queries#readonly, we must use POST for such queries, so it should also affect the `use_post` flag.